### PR TITLE
fix preserveScrollPosition for ember 3.6+

### LINKED ISSUE #197: 1.1.0 breaks preserveScrollPosition
I have a controller which sets preserveScrollPosition to true. If I install version 1.1.0 of this add-on and enter that route, the page scrolls to the top.

If I downgrade to 1.0.2, then the scroll position stays at the position where it has been previously.

It looks like this is related to how ember +3.6 is being handled differently since version 1.1.0
https://github.com/dollarshaveclub/ember-router-scroll/blob/24799d770e200a605e001dd79cec4169e353afb1/addon/index.js#L63 
in `get(transition, 'handler.controller.preserveScrollPosition')` handler is undefined, therefore we are not able to retrieve `preserveScrollPosition` from the controller

### PR DESCRIPTION
**Pull request by st-h**
fixes #197 

opened a [topic](https://discuss.emberjs.com/t/is-this-an-accepted-not-private-way-to-get-the-controller-from-a-transition/16551) over at the ember forum to get some information if this is ok or if it actually might be touching private apis we should not be using

### CODE REVIEW

File: addon/index.js [hunk 1]
```diff
@@ -60,7 +60,7 @@ let RouterScrollMixin = Mixin.create({
 
     let preserveScrollPosition;
     if (gte('3.6.0-beta.1')) {
-      preserveScrollPosition = get(transition, 'handler.controller.preserveScrollPosition');
+      preserveScrollPosition = transition.intent.router.currentRouteInfos.some((routeInfo) => get(routeInfo, 'route.controller.preserveScrollPosition'));
```
Comment (snewcomer) [inline addon/index.js]: do we want safe-get with `get(transition, 'intent....')`?

**Comment by snewcomer** [comment]
Note - this is similar, but different work to #195

File: addon/index.js (hunk 1 above)
Comment (st-h) [reply addon/index.js]: yes, that sounds reasonable

**Comment by st-h** [comment]
> Note - this is similar, but different work to #195

I believe this is a misunderstanding. This PR fixes a regression introduced in 1.1.0, which causes the `preserveScrollPosition` property to have no effect. This is especially bad, as this can impact the native behaviour of mobile browsers, where an active input element is automatically scrolled into the viewport.

However #195 adds a new feature that allows to control scrolling behaviour via a new service. #195 never touches any code related to the regression.

**Comment by snewcomer** [comment]
Thank you for the PR @st-h

**Comment by st-h** [comment]
@snewcomer I received an interesting comment on the ember forum regarding the use of private apis: https://discuss.emberjs.com/t/is-this-an-accepted-not-private-way-to-get-the-controller-from-a-transition/16551/2

**Comment by snewcomer** [comment]
new PR!

**Comment by st-h** [comment]
@snewcomer yes, this absolutely should be a separate PR. However, I am reluctant what the best option actually is. On the one hand, I think the service from #195 would be a clean way to solve this. On the other hand, this probably makes excluding a single route more complex as one has to set the flag on the transition and reset it when transitioning away. At least there should not be any downside (except being a breaking change) in moving the property from the controller to the route. But then lastly, currently we do not really gain anything in making that change as long as things work.

**Comment by st-h** [comment]
@snewcomer do you have plans/a date for a release? 1.0.2 is not affected by this issue, however it gives us plenty of deprecation warning, which should be fixed in 1.1.0.

**Comment by snewcomer** [comment]
Yep! Hopefully will have publish right sometimes this week so...relatively soon.  Thanks for the reminder

**Pull request merged**